.github/workflows/build-and-test.yml

```diff
@@ -44,7 +44,7 @@ jobs:
         run: ./gradlew --build-cache :service:build -x test
 
       - name: Upload spotbugs results
-        uses: github/codeql-action/upload-sarif@main
+        uses: github/codeql-action/upload-sarif@v2
```

Comment:
Works around https://github.com/github/codeql-action/issues/1671. I'm not sure pulling from main is really such a good idea anyway….